SpringBox/pair_interactions.py

```diff
@@ -2,57 +2,75 @@
 import numpy as np
 from scipy.spatial.distance import pdist, squareform
 
+
 def active_particles(pXs, prv_acc, activation_fn, _config):
     # Particles that were previously active, now can loose their activation with a rate of activation_decay_rate
     prv_acc = prv_acc * (np.random.rand(len(prv_acc)) > _config['activation_decay_rate'] * _config['dt'])
     # Or particles can become active when they enter the light activated area
     new_acc = activation_fn(pXs)
     return np.maximum(prv_acc, new_acc)
 
+
 @numba.jit
-def spring_forces(acc,pXs , Dij, _config):
+def spring_forces(acc, pXs, Dij, Drxns, _config):
     rhs = np.zeros_like(pXs)
-    n_part=_config['n_part']
-    k=_config['spring_k']
-    r0=_config['spring_r0']
-    Iij = Dij * np.outer(acc,acc)
-    Iij = (Iij>_config['spring_lower_cutoff']) * (Iij<_config['spring_cutoff'])
+    n_part = _config['n_part']
+    k = _config['spring_k']
+    r0 = _config['spring_r0']
+    Iij = Dij * np.outer(acc, acc)
+    Iij = (Iij > _config['spring_lower_cutoff']) * (Iij < _config['spring_cutoff'])
     for i in range(n_part):
-        for j in range(i+1,n_part):
-            if Iij[i,j]!=0:
-                rhs[i] += -k*((Dij[i,j]-r0)/Dij[i,j])*(pXs[i]-pXs[j])
-                rhs[j] += +k*((Dij[i,j]-r0)/Dij[i,j])*(pXs[i]-pXs[j])
+        for j in range(i + 1, n_part):
+            if Iij[i, j] != 0:
+                rhs[i] += -k * ((Dij[i, j] - r0) / Dij[i, j]) * (Drxns[i][j])
+                rhs[j] += +k * ((Dij[i, j] - r0) / Dij[i, j]) * (Drxns[i][j])
     return rhs
+
+
 @numba.jit
-def LJ_forces(acc,pXs , Dij, _config):
-    n_part=_config['n_part']
+def LJ_forces(acc, pXs, Dij, _config):
```

Comment:
Since you have the code ready, could you also make the same changes here? Then the code should always work the same.